lib/elixir/test/elixir/option_parser_test.exs

```diff
@@ -256,6 +256,12 @@ defmodule OptionParserTest do
            == {[foo: true, boo: "-"], ["-"], []}
   end
 
+  test "allow nonexistent atoms" do
+    opts = [switches: [a: :string], allow_nonexistent_atoms: true]
```

Comment:
I think we not testing this option here, as we have all atoms defined in the test.
Also it'd be better to use more descriptive option names, like we use "--option-key-does-not-exist" in other place. Perhaps we can use "--option-key-creates-atom".